sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/SQLHadoopMapReduceCommitProtocol.scala

```diff
@@ -32,10 +35,14 @@ import org.apache.spark.sql.internal.SQLConf
 class SQLHadoopMapReduceCommitProtocol(
     jobId: String,
     path: String,
-    dynamicPartitionOverwrite: Boolean = false)
+    dynamicPartitionOverwrite: Boolean = false,
```

Comment:
Can we make this parameter a collection/set/map. ?? making it so keeps the interface fixed and also extendable for future.
Suppose we have more such behaviours to be added (which I am sure will happen), we cannot create new interfaces and break the contract each time(though we try not to).
Having a Key-Value pair of configuration will be much easier for such extensions.

Reply:
Done for this.